src/canvas.py

```diff
@@ -268,6 +270,17 @@ def items(self, items: list):
                 axis_legend.remove()
             if used:
                 self._legend_axis = axis
+                used_directions += directions
+
+        used_directions = list(set(used_directions))  # Remove duplicates
+        for direction in ["bottom", "left", "top", "right"]:
```

Comment:
Basically this entire loop could be skipped if the frame is turned on. Should be easy to adapt to that.